bot/commands/imagesearch.js

```diff
@@ -16,19 +16,31 @@ exports.init = function () {
 
             otterbot.log('Searching for an image: ' + message);
 
-            request(url, function (err, res, body) {
-                if (!err && res.statusCode == 200) {
-                    var images = JSON.parse(body)["responseData"]["results"];
+            var fetchImage = function(images) {
```

Comment:
In general I think having a "fetch this url and give me back a promise I can call .success and .error on" is useful.

In this particular instance there is a wrinkle with using a method like that because we also need to check the content type. That results in having the code change into something like....

```Javascript
request(url).success(function(headers) {
    if(_.contains(headers['content-type'], "image/")) {
         // this is actually the success case
    } else {
        // this is an error case (in the success block, ick)
    }
}).error(function() {
     // this is the main error case
})
```

Not sure, at least in this case, it's much better.

I'd say lets ship this and then we can add Helper.request() with all the promise magic and find the places to use it.